tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+        label: 'Event 2 Today!'
+    },
+    {
+        date: new Date( 2022, 8, 17 ),
+        label: 'Event 3 Today!'
+    },
+    {
+        date: new Date( 2022, 8, 20 ),
+        label: 'Event 1 Another Day!'
+    }
+]);
+
+moduleForComponent( 'sl-calendar', 'Integration | Component | sl calendar', {
+    integration: true
+});
+
+test( 'Setting currentYear and currentMonth modifies the view correctly', function( assert ) {
+
+    const currentYear = 2025;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 1;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'January ' + currentYear,
+        'Current month and year are set correctly in the view'
+    );
+});
+
+test( 'Setting dateValuePath modifies the view correctly', function( assert ) {
+
+    const modifiedValuePath = Ember.A([
+        {
+            modifiedDatePath: new Date( 2022, 8, 17 ),
+            label: 'Today!'
+        }
+    ]);
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', modifiedValuePath );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            dateValuePath='modifiedDatePath'
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        17,
+        'Active day is set correctly'
+    );
+});
+
+test( 'Active day is set correctly', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    const emptyContent = Ember.A( [ {} ] );
+    this.set( 'content', emptyContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        '',
+        'No active day is set'
+    );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        17,
+        'Active day is set correctly'
+    );
+
+    this.set( 'content', multipleTestContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
```

Comment:
Added double quotes. The action="testAction" is needed because we are firing the action and verifying that the object contents are passed through to the action.